### ISSUE TITLE
space acres 0.1.10 deletes config.json with farm size typos

### ISSUE THREAD

**Issue by madmax-drboz** [subspace/space-acres#168]
OS is Windows 10 and all software is up-to-date.
file in question that prevents Space-Acres from load is at: C:\Users\NASA\AppData\Local\space-acres\config.json

A previously added farm specified for 300GiB was manually edited to 300GB, which is smaller in the GUI. The GUI asks you to reload space acres. This causes Space acres to completely empty the config.json file upon relaunching. Using the .msi installer to repair or doing a completely clean install does not restore config.json back to default. End user has to manually copy and fix the config.json. The file path also needs to be added with double slash as in D://farm to make the config work per syntax that is not the same as the linux version. There's some area to get lost here.

Another related issue is when someone types a farm size bigger than the actual drive such as the drive itself maybe 800GB but I mistakenly typed 800TB which prevents Space Acres from loading at all. One would need to find the config.json and fix the file before the GUI is able to load again.

In short, if the end user types a number bigger than the drive size for a new plot, or try to retroactively change an existing plot size to a smaller size, the GUI will no longer load up.

**Comment by madmax-drboz** [comment]
there's also another funny "bug" with space acres
which is if the space acres window is maximized at all, it runs up 30% on a 3070Ti GPU, which is more usage than netflix or youtube 4k. I assume this is related to the rendering of the color coded plot status.
as soon as you minimize the window, GPU usage drops to 0

**Comment by nazar-pc** [comment]
What you write is a bit confusing and self-conflicting.

> A previously added farm specified for 300GiB was manually edited to 300GB, which is smaller in the GUI.

By manually you mean opening config file and editing it yourself rather than using GUI?

> The GUI asks you to reload space acres. This causes Space acres to completely empty the config.json file upon relaunching.

Manual file editing doesn't result in application asking to restart and I have not heard file being empty on restart, this doesn't make either.

> Using the .msi installer to repair or doing a completely clean install does not restore config.json back to default.

It doesn't, it just repairs the app, not its configs, at least right now.

> End user has to manually copy and fix the config.json. The file path also needs to be added with double slash as in D://farm to make the config work per syntax that is not the same as the linux version. There's some area to get lost here.

You should never need to edit that file manually unless you know what you are during, including escaping rules in JSON.

> Another related issue is when someone types a farm size bigger than the actual drive such as the drive itself maybe 800GB but I mistakenly typed 800TB which prevents Space Acres from loading at all. One would need to find the config.json and fix the file before the GUI is able to load again.
>
> In short, if the end user types a number bigger than the drive size for a new plot, or try to retroactively change an existing plot size to a smaller size, the GUI will no longer load up.

One problem per issue, please. Don't mix multiple at once, it makes things confusing. This particular one is already tracked in https://github.com/subspace/space-acres/issues/115

> there's also another funny "bug" with space acres
which is if the space acres window is maximized at all, it runs up 30% on a 3070Ti GPU, which is more usage than netflix or youtube 4k. I assume this is related to the rendering of the color coded plot status.
as soon as you minimize the window, GPU usage drops to 0

Again, create separate issue for separate problems, don't put everything together.

---

It would be helpful to clearly describe step by step what you did, what your expectations where and what happened instead. The description is long and I don't understand what is going on except that you've got an issue.

**Comment by nazar-pc** [comment]
There were some fixes in recent versions, please provide more details if this is still happening
